src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -645,7 +646,19 @@ protected void runSpooler() {
                         }
                     }
 
-                    CompletableFuture<Integer> future = publishSingleSpoolerMessage();
+                    // Select connection with minimum time to wait before publishing the next message
+                    AwsIotMqttClient connection = getConnection(false);
+                    long minimumWaitTimeMicros = connection.getThrottlingWaitTimeMicros();
+                    for (AwsIotMqttClient client : connections) {
+                        long waitTime = client.getThrottlingWaitTimeMicros();
+                        if (waitTime < minimumWaitTimeMicros) {
+                            connection = client;
+                            minimumWaitTimeMicros = waitTime;
+                        }
+                    }
+                    TimeUnit.MICROSECONDS.sleep(minimumWaitTimeMicros);
```

Comment:
can you add a comment here about why the sleep was added here. It isn't intuitive to have to sleep here while also still calling `acquire` in the client.

Did you keep those because some other component can use the MQTT client and publish while the spooler is sleeping?

Reply:
Acquire is what actually takes the tokens from the bucket. Prior to that I'm just querying when we can next take a token, but not actually taking it out since I need to query multiple limiters to find the minimum wait. Since we're doing the sleeping, the acquire is guaranteed to not block.

Reply:
right - can you add that as a comment to here in the source?

Reply:
Yes I'm adding that

Reply:
Updated.